libraries/resources/api200/resource_provider.rb

```diff
@@ -0,0 +1,21 @@
+# (c) Copyright 2016 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API200
+    # Base class for API200 resource providers
+    class ResourceProvider < OneviewCookbook::ResourceProvider
```

Comment:
Yes. I didn't include the one for API300 and the variants, but yes, the concept is the same. We'd add the default class in there. Exactly like the Resource class for the SDK